### PR TITLE
*: Upgrade fmtlib and print enum value explicitly

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/8331

Problem Summary:

The latest version of fmtlib disables implicit casts from `enum`/`enum class` to `int`, which leads to compile error. We must resolve those errors with `magic_enum::enum_name` to print out the enum nam or `fmt::underlying` to print out the int value.

Reference: https://github.com/fmtlib/fmt/issues/3671

An example of the compile error raised by upgraded fmtlib
```
...  -c x/tiflash-master/dbms/src/Flash/tests/gtest_collation.cpp
In file included from x/tiflash-master/dbms/src/Flash/tests/gtest_collation.cpp:16:
In file included from x/tiflash-master/dbms/src/TestUtils/ExecutorTestUtils.h:20:
In file included from x/tiflash-master/dbms/src/TestUtils/ExecutorSerializer.h:17:
In file included from x/tiflash-master/dbms/src/Common/FmtUtils.h:17:
In file included from x/tiflash-master/contrib/fmtlib-cmake/../fmtlib/include/fmt/compile.h:11:
In file included from x/tiflash-master/contrib/fmtlib-cmake/../fmtlib/include/fmt/format.h:49:
x/tiflash-master/contrib/fmtlib-cmake/../fmtlib/include/fmt/core.h:2548:45: error: implicit instantiation of undefined template 'fmt::detail::type_is_unformattable_for<tipb::ExecType, char>'
    type_is_unformattable_for<T, char_type> _;
                                            ^
x/tiflash-master/contrib/fmtlib-cmake/../fmtlib/include/fmt/core.h:2611:23: note: in instantiation of function template specialization 'fmt::detail::parse_format_specs<tipb::ExecType, fmt::detail::compile_parse_context<char>>' requested here
        parse_funcs_{&parse_format_specs<Args, parse_context_type>...} {}
                      ^
x/tiflash-master/contrib/fmtlib-cmake/../fmtlib/include/fmt/core.h:2740:47: note: in instantiation of member function 'fmt::detail::format_string_checker<char, tipb::ExecType>::format_string_checker' requested here
      detail::parse_format_string<true>(str_, checker(s));
                                              ^
x/tiflash-master/dbms/src/Flash/tests/gtest_collation.cpp:316:46: note: in instantiation of function template specialization 'fmt::basic_format_string<char, tipb::ExecType &>::basic_format_string<char[22], 0>' requested here
            auto exception_str = fmt::format("Unhandled executor {}", type);
                                             ^
x/tiflash-master/contrib/fmtlib-cmake/../fmtlib/include/fmt/core.h:1554:45: note: template is declared here
template <typename T, typename Char> struct type_is_unformattable_for;
```

### What is changed and how it works?

* Upgrade fmtlib to [v10.1.1](https://github.com/fmtlib/fmt/releases/tag/10.1.1)
* When we are sure the enum value has corresponding name, we use `magic_enum::enum_name` to cast it to string
* When we are not sure the enum value has corresponding name, we use `fmt::underlying` to cast it to int
  * for example, the enum type defined in kvproto/tipb
* For some classes, we implement a `format_as` function to make it printable by fmtlib
  * `MppVersion`/`MPPDataPacketVersion`/`StringRef`/`SettingInt`/`SettingFloat`/`SettingDouble`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/TestUtils/tests/gtest_mock_executors.cpp [hunk 1]
```diff
@@ -356,10 +356,11 @@ try
...
     {
-        String expected = "window_2 | partition_by: {(<1, String>, desc: false)}}, order_by: {(<0, String>, desc: "
-                          "true)}, func_desc: {row_number()}, frame: {start<2, false, 0>, end<2, false, 0>}\n"
-                          " sort_1 | isPartialSort: true, partition_by: {(<0, String>, desc: false)}\n"
-                          "  table_scan_0 | {<0, String>, <1, String>}\n";
+        String expected
+            = "window_2 | partition_by: {(<1, String>, desc: false)}}, order_by: {(<0, String>, desc: "
+              "true)}, func_desc: {row_number()}, frame: {start<CurrentRow, false, 0>, end<CurrentRow, false, 0>}\n"
+              " sort_1 | isPartialSort: true, partition_by: {(<0, String>, desc: false)}\n"
+              "  table_scan_0 | {<0, String>, <1, String>}\n";
```
Comment (JaySon-Huang) [inline dbms/src/TestUtils/tests/gtest_mock_executors.cpp:363]: This is cause by the changes in `TestUtils/ExecutorSerializer.cpp`

**Pull request merged**